src/ensembl/doc/architecture/decisions/0018-indexeddb.md

```diff
@@ -34,26 +34,52 @@ There are several features of IndexedDB that make it a particularly attractive s
 - Safari is known to break IndexedDB (for example, there is a bug in Safari, both desktop and iOS version, that prevents the immediate use of IndexedDB when the browser is opened). We hope that this will not be a significant concern in the future. Also, to be fair, Safari is also known to have broken LocalStorage.
 
-## To consider
-There exists a small library, [idb-keyval](https://www.npmjs.com/package/idb-keyval), which provides a simple API to use IndexedDB just as a key-value storage, combining the simplicity of LocalStorage with the power of IndexedDB (asynchronicity and different data types). It removes the complexity of managing IndexedDB; however, in our case that would return us to the question of how to handle updatets to the schemas of stored data, which can otherwise be solved through IndexedDB versioning. 
+## Practical considerations: database versioning 
+Consider the following code snippet of using IndexedDB with the `idb` library:
+
+```ts
+import { openDB } from 'idb';
+
+const DB_NAME = 'my-database';
+const DB_VERSION = 1;
+
+openDB(DB_NAME, DB_VERSION, {
+  upgrade(db) {
+    if (!db.objectStoreNames.contains('contact-forms')) {
+      db.createObjectStore('contact-forms');
+    }
+  }
+});
+```
+
+When a database connection is opened, either a new database gets created, if this is the first time that this code runs in the user's browser, or a connection to an already existing database is established. In the example above, the version of the database is going to be `1`, and it will contain a single object store specifically to store contact form data.
+
+Once a database has been created, the only way to add or remove an object store is through the change of the database version. The database version can only be changed in the forward direction, i.e. every subsequent database version must be greater than the previous one; it is impossible to downgrade a higher-versioned database to a lower version (which means, regarding the snippet above, that if the user already has the "my-database" database at version 2, the snippet will result in an error).
+
+When the database version is incremented, IndexedDB will run an appropriate callback, in which the state of the database before the upgrade can be inspected, and appropriate steps can be taken to migrate the currently stored data. By default, if no specific actions are taken during a version bump, the already saved records will remain, unmodified, in the new version of the database.
+
+The update of the database version provides an opportunity for the developer to migrate the already saved data to a new shape as required by the updates to the code. It is important to remember, however, is that it is impossible to predict which database version any given user will already have; so the migration strategy can get complicated.
```

Comment:
There is an extra `is` in `is that it is impossible`

Reply:
✅